cdap-app-fabric/src/main/java/co/cask/cdap/app/mapreduce/MRJobClient.java

```diff
@@ -82,6 +82,10 @@ public MRJobInfo getMRJobInfo(Id.Run runId) throws IOException, NotFoundExceptio
     JobStatus thisJob = findJobForRunId(jobs, runId);
 
     RunningJob runningJob = jobClient.getJob(thisJob.getJobID());
+    if (runningJob == null) {
+      LOG.warn("JobClient returned null for JobId: '{}'", thisJob.getJobID());
+      throw new NotFoundException(runId);
```

Comment:
This will be thrown when the runid is available but the jobClient could not return the corresponding `RunningJob`. Should it be `IllegalStateException`?